calico_kubernetes/calico_kubernetes.py

```diff
@@ -328,10 +362,33 @@ def _apply_rules(self, profile_name):
             print("Error: Could not apply rules. Profile not found: %s, exiting" % profile_name)
             sys.exit(1)
 
-        rules = self._generate_rules()
-        profile_json = self._generate_profile_json(profile_name, rules)
-        profile.rules = Rules.from_json(profile_json)
-        self._datastore_client.profile_update_rules(profile)
+        inbound_rules, outbound_rules = self._generate_rules(pod)
+
+        print "Removing Default Rules"
+
+        # TODO: This method is append-only, not profile replacement, we need to remove default rules
+        # TODO: remove calicoctl calls (x5)
+        try:
```

Comment:
This is a bit weird.

We should know exactly what rules are created by default, correct?

If you want to be resilient to changes, you might want to do this in a while loop, and break when you hit the specific error thrown when there are no more rules.